### ISSUE TITLE
Runtime exception in when statement with string using native but not in JVM

### ISSUE THREAD

**Issue by zero9178** [JetBrains/kotlin-native#1941]
Following code works in kotlin running in JVM but throws an exception in native.
```
fun main(args: Array<String>) {
    when(args.first()) {
        "install" -> print("Success")
        else -> {
            print("Error")
        }
    }
}
```
Doesn't matter what the program arguments are it will throw the same exception (except if there are 0).
The exception in native is:
```
Uncaught exception from Kotlin's main: kotlin.RuntimeException: Unexpected receiver type: kotlin.String
        at kfun:kotlin.Throwable.<init>(kotlin.String?)kotlin.Throwable (0000000000403150)
        at kfun:kotlin.Exception.<init>(kotlin.String?)kotlin.Exception (0000000000403070)
        at kfun:kotlin.RuntimeException.<init>(kotlin.String?)kotlin.RuntimeException (0000000000402f90)
        at kfun:konan.internal.ThrowInvalidReceiverTypeException(kotlin.reflect.KClass<#STAR>)kotlin.Nothing (0000000000
401ea0)
        at kfun:main(kotlin.Array<kotlin.String>) (00000000004017a0)
        at EntryPointSelector (0000000000401790)
        at Konan_start (00000000004016c0)
        at Konan_run_start (0000000000401620)
        at Konan_main (00000000004015f0)
        at __tmainCRTStartup (00000000004011b0)
        at mainCRTStartup (0000000000401510)
        at  (00007ffa93523020)
        at  (00007ffa93ce1410)
```

If I am doing something wrong instead please tell me!

**Comment by olonho** [comment]
Nope, this is a compiler bug, to work around - use `--disable divirtualization` switch.

**Comment by zero9178** [comment]
Thank you that worked yes! But I believe you ment "devirtualization"

**Comment by olonho** [comment]
Fixed with cc28f81ff0dd9c3ceb454c0b89b2568b1f3d83ee
